diesel/src/result.rs

```diff
@@ -414,3 +414,32 @@ impl fmt::Display for UnexpectedEndOfRow {
 }
 
 impl StdError for UnexpectedEndOfRow {}
+
+/// An error occurred while deserializing a field
+#[derive(Debug)]
+pub struct DeserializeFieldError {
```

Comment:
```suggestion
#[non_exhaustive]
pub struct DeserializeFieldError {
```

We don't  want that users can construct this error type. We also want to be able to add fields later on.

Reply:
Just to check - you want me to remove the `pub` here?

Reply:
`pub` should stay there, as you likely want to be able to access these fields from your application as wel.